src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -32,6 +34,29 @@
     private static Bundle gCachedExtras = null;
     private static boolean gForeground = false;
 
+    @Override
+    public void initialize(CordovaInterface cordova, CordovaWebView webView) {
+        super.initialize(cordova, webView);
+        gForeground = true;
+
+        if (isAmazonDevice()) {
```

Comment:
I just like to have the most likely option at the top of my if then else chain. What you've done is fine, just a personal preference.